app/models/accessionable/base.rb

```diff
@@ -79,16 +79,82 @@ def label_scope
   class Tag # rubocop:todo Style/Documentation
     attr_reader :value
 
+    # Value serialization for accessioning XML generation
+    # It will return the same value without any changes and it applies
+    # to all tags
+    class FieldSerializer
+      NOT_COLLECTED = 'not collected'
+      NOT_PROVIDED = 'not provided'
+      RESTRICTED_ACCESS = 'restricted access'
+
+      OTHER_DEFAULT_SETTINGS = [NOT_COLLECTED, NOT_PROVIDED, RESTRICTED_ACCESS].freeze
+
+      def value_for(value)
+        value
+      end
+      def applies_to?(_name)
+        true
+      end
+      def incorrect_format_value
+        NOT_COLLECTED
+      end
+    end
+
+    # Value serialization for country of origin in accessioning XML generation
+    # It will return a valid country of origin or 'not collected' if nothing provided or invalid
+    # It also allow other config settings for the XML service like the list defined inside OTHER_DEFAULT_SETTINGS
+    class FieldCountryOfOrigin < FieldSerializer
+      def value_for(value)
+        return value if OTHER_DEFAULT_SETTINGS.include?(value)
+        return incorrect_format_value unless Insdc::Country.find_by(name: value)
+        value
+      end
+
+      def applies_to?(name)
+        name == :country_of_origin
+      end
+    end
+
+    # Value serialization for collection date in accessioning XML generation
+    # It will return a valid collection date or 'not collected' if nothing provided or invalid
+    # It also allow other config settings for the XML service like the list defined inside OTHER_DEFAULT_SETTINGS
+    # NB: this regexp is defined in <https://www.ebi.ac.uk/ena/browser/api/xml/ERC000011>
+    class FieldCollectionDate < FieldSerializer
+      # rubocop:disable Layout/LineLength
```

Comment:
Is this a regexp to check for a valid date?

Reply:
Yes, to me it seems it has been automatically generated. It is what they specified in https://www.ebi.ac.uk/ena/browser/view/ERC000011